### PR TITLE
0.13 example update: local-mode-k8s

### LINKED ISSUE #3752: 0.13 example update: local-mode-k8s

### PR DESCRIPTION
**Pull request by vvagaytsev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #3752

**Special notes for your reviewer**:

### CODE REVIEW

File: examples/local-mode-k8s/README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
-# Local mode for `kubernetes` modules
+# Local mode for `kubernetes` action type
 
 A very basic demo project for Garden [local mode](../../docs/guides/running-service-in-local-mode.md) for `kubernetes`
-module type.
+action type.
 
 This project is based on the [demo-project](../demo-project). The only difference is that this one has the `backend`
-service defined as a `kubernetes` module with the `backend-image` container module used as a source Docker image.
+service defined as an action of type `kubernetes` with the `backend-image` container used as a source Docker image.
```
Comment (Orzelius) [inline examples/local-mode-k8s/README.md:7]:
```suggestion
service defined as an action of type `kubernetes` with the `backend-image` container build action used as a source Docker image.
```

File: examples/local-mode-k8s/README.md [hunk 2]
```diff
@@ -33,14 +33,5 @@ garden deploy --local=*
 
 Alternatively, you can also run the local mode using the `garden dev` command:
```
Comment (Walther) [inline examples/local-mode-k8s/README.md:34]: nit: could probably remove this line too, if the shell examples are removed below it

Comment (vvagaytsev) [reply examples/local-mode-k8s/README.md:34]: Ah, right! Good catch! :)

**Review by Orzelius — APPROVED** [review]
had one comment

File: examples/local-mode-k8s/README.md (hunk 2 above)
Comment (Orzelius) [reply examples/local-mode-k8s/README.md]: No I think it should stay, but maybe add a small guide on how to do the same thing with the new dev command

Comment (vvagaytsev) [reply examples/local-mode-k8s/README.md]: Afaik, the new dev command is not fully implemented yet, and now delegates to `garden serve`. We can add the `dev`-command guides and examples later.

**Review by Walther — CHANGES_REQUESTED** [review]
Looking mostly good! 👍

Running `garden test` seems to fail on the integration tests:

```
+ expected - actual

      {
-  "message": "Backend says: 'Hello from Dockerized Go!'"
+  "message": "Backend says: 'Hello from 1st Go!'"
      }
```

Not sure if this is an error in the config (ingress?) or the test code though.

**Comment by vvagaytsev** [comment]
Thanks, @Walther! The integration tests were broken in the `main` branch as well, and it's fixed in #3770. I'll cherry-pick that fix to this PR.

**Review by Walther — APPROVED** [review]
great work! ✨

**Pull request merged**